src/openassetio-core-c/include/openassetio/c/namespace.h

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <openassetio/export.h>  // For OPENASSETIO_VERSION
+
+/**
+ * @addtogroup CAPI C API
+ * @{
+ */
+
+/**
+ * Prefix `openassetio_vX_Y_` to a given symbol name, where `X_Y` is
+ * the current API version.
+ *
+ * @hideinitializer
+ */
+#define OPENASSETIO_NS(symbol) OPENASSETIO_NS_WITH_VER(OPENASSETIO_VERSION, symbol)
+
+/**
+ * @}
+ */
+
+// Utility to work around token pasting macro expansion rules.
+#define OPENASSETIO_NS_WITH_VER(ver, symbol) OPENASSETIO_NS_WITH_VER_IMPL(ver, symbol)
+// Utility to work around token pasting macro expansion rules.
+#define OPENASSETIO_NS_WITH_VER_IMPL(ver, symbol) openassetio_##ver##_##symbol
```

Comment:
Both of those are conceptually private. I tried adding the `@private` for Doxygen, but it didn't make any difference. What did work was putting the `@}` above these so they don't appear in the docs.  Perhaps it would be worth adding `@private` anyway, for people browsing the source?

Reply:
[Unfortunately](https://en.cppreference.com/w/cpp/language/identifiers)

>  * the identifiers with a double underscore anywhere are reserved;
>    ...
>  * the identifiers that begin with an underscore are reserved in the global namespace. 

Technically we could, since it's unlikely our identifiers will conflict with any standard library!

Reply:
It's fine as-is. Maybe worth expanding the docs as to the workaround that is being used, and an explicit sentence saying these are part of that, and shouldn't be used directly?

Reply:
Updated doxygen docs in 6503bd53